File: library/src/main/java/com/smartnsoft/smartappupdate/bo/UpdatePopupInformations.java
package com.smartnsoft.smartappupdate.bo;

import java.io.Serializable;

import com.smartnsoft.smartappupdate.SmartAppUpdateManager.UpdatePopupType;

/**
 * @author Adrien Vitti
 * @since 2018.01.23
 */

public final class UpdatePopupInformations
    implements Serializable
{

  public String title;

  public String imageURL;

  public String updateContent;

  public String changelogContent;

  public String actionButtonLabel;

  public String deepLink;

  public String packageName;

  public long versionCode;

  @UpdatePopupType
  public int updatePopupType;

}

File: library/src/main/java/com/smartnsoft/smartappupdate/bo/UpdatePopupInformations.kt
package com.smartnsoft.smartappupdate.bo

import com.smartnsoft.smartappupdate.SmartAppUpdateManager.UpdatePopupType
import java.io.Serializable

/**
 * @author Adrien Vitti
 * @since 2018.01.23
 */

data class UpdatePopupInformations(
    var title: String? = null,
    var imageURL: String? = null,
    var updateContent: String? = null,
    var changelogContent: String? = null,
    var actionButtonLabel: String? = null,
    var deepLink: String? = null,
    var packageName: String? = null,
    var versionCode: Long = 0,
    @UpdatePopupType
    var updatePopupType: Int = 0
) : Serializable
